stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/security/KeyStore.java

```diff
@@ -0,0 +1,256 @@
+        if (certificate == null) {
+            return null;
+        }
+        return certificate.getPublicKey();
+    }
+
+    /**
+     * Get the certificate for a given key alias from keyStore.
+     *
+     * @param alias the alias name
+     * @return certificate corresponding to the alias
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), or this operation fails for some other reason
+     */
+    public Certificate getCertificate(String alias) throws KeyStoreException {
+        try {
+            return keyStore.getCertificate(alias);
+        } catch (java.security.KeyStoreException e) {
+            throw new KeyStoreException("Failed to load certificate: " + alias, e);
+        }
+    }
+
+    /**
+     * Get the default private key of the service.
+     *
+     * @return default private key
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), if the algorithm for recovering the entry cannot be found,
+     *                           Keystore configuration does not contain
+     *                           the information needed to recover the key (e.g. wrong password)
+     */
+    public PrivateKey getDefaultPrivateKey() throws KeyStoreException {
+        ConfigRegistry configRegistry = ConfigRegistry.getInstance();
+        char[] keyStorePassword = configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_PASSWORD)
+                .toCharArray();
+        String keyAlias = configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_ALIAS);
+        return getPrivateKey(keyAlias, keyStorePassword);
+    }
+
+    /**
+     * Get the default public key for the service.
+     *
+     * @return default public key
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), or this operation fails for some other reason
+     */
+    public PublicKey getDefaultPublicKey() throws KeyStoreException {
+        String keyAlias = ConfigRegistry.getInstance().getInstanceConfigValue(KEY_STORE_CONFIG, KEY_ALIAS);
+        return getPublicKey(keyAlias);
+    }
+
+    /**
+     * Get the default certificate for the service.
+     *
+     * @return certificate corresponding to the alias
+     * @throws KeyStoreException if the keystore has not been initialized
+     *                           (loaded), or this operation fails for some other reason
+     */
+    public Certificate getDefaultCertificate() throws KeyStoreException {
+        String keyAlias = ConfigRegistry.getInstance().getInstanceConfigValue(KEY_STORE_CONFIG, KEY_ALIAS);
+        return getCertificate(keyAlias);
+    }
+
+    private void loadKeyStore() {
+        ConfigRegistry configRegistry = ConfigRegistry.getInstance();
+        String keyStoreLocation = configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_STORE_LOCATION);
+        if (keyStoreLocation != null) {
```

Comment:
Can this be empty?

Reply:
This should not be null. This is used until registry API provides a method to check whether main config block exists.